File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{

    string text = get_string("Text: ");

    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    float L = letters / words * 100;
    float S = sentences / words * 100;

    float rough = 0.0588 * L - 0.296 * S - 15.8;

    int index = round(rough);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %d\n", index);
    }

}

int count_letters(string text)
{
    int total = 0;
    int n = strlen(text);

    for (int i = 0; i < n; i++)
    {
        if (isalpha(text[i]))
        {
            total += 1;
        }
    }

    return total;
}

int count_words(string text)
{
    int total = 0;
    int n = strlen(text);

    for (int i = 0; i < n; i++)
    {
        if (text[i] == ' ')
        {
            total += 1;
        }
    }
    total += 1;
    return total;
}

int count_sentences(string text)
{
    int total = 0;
    int n = strlen(text);

    for (int i = 0; i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            total += 1;
        }
    }

    return total;
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0
words = 1
sentences = 0

for i in range(len(text)):
    if text[i].isalpha():
        letters += 1

for i in range(len(text)):
    if text[i] == " ":
        words += 1

for i in range(len(text)):
    if text[i] == "." or text[i] == "?" or text[i] == "!":
        sentences += 1
        
L = float(letters / words * 100)
S = float(sentences / words * 100)

rough = .0588 * L - .296 * S - 15.8

index = round(rough)

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")
